bookkeeper-common/src/main/java/org/apache/bookkeeper/common/util/OrderedExecutor.java

```diff
@@ -391,6 +391,10 @@ protected OrderedExecutor(String baseName, int numThreads, ThreadFactory threadF
             ExecutorService thread = createSingleThreadExecutor(
                     new ThreadFactoryBuilder().setNameFormat(name + "-" + getClass().getSimpleName() + "-" + i + "-%d")
                     .setThreadFactory(threadFactory).build());
+            SingleThreadExecutor ste = null;
+            if (thread instanceof SingleThreadExecutor) {
```

Comment:
We can registerMetrics here instead of the instanceOf check

Reply:
Do you mean that remove the check `if (thread instanceof SingleThreadExecutor)` directly ?
But createSingleThreadExecutor() is also triggered in OrderedScheduler. In OrderedScheduler, createSingleThreadExecutor() do not return SingleThreadExecutor, so I keep the check here.